contentful/apps
# Add segment analytics tracking to GA4

## Purpose

We want to track user engagement with the new GA4 app. This PR introduces tracking analytics via Segment

## Approach

* Use https://github.com/segmentio/analytics-next/tree/master/packages/browser#readme installed via package instead of adding the snippet to the top (as outlined https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/quickstart/)
* Set up as content provider paired with hook that intializes the singleton client and identifies the user and is then readily available to other components -- pretty similar to `useSDK`

### Out of scope

* Meaningful tracking and page details. This PR is just intended to get the basic tool working

### TODO

* [ ] Set up a production source and write key in circle CI

## Testing steps

1. Visit the app config and an entry where the GA sidebar loads
2. Verify that the data shows up in the staging source

## Breaking Changes



## Dependencies and/or References



## Deployment
